test/integration/BUILD

```diff
@@ -876,6 +876,21 @@ envoy_cc_test(
     ],
 )
 
+envoy_cc_test(
+    name = "extension_discovery_integration_test",
+    srcs = ["extension_discovery_integration_test.cc"],
+    tags = ["fails_on_windows"],
+    deps = [
+        ":http_integration_lib",
+        "//source/extensions/filters/http/rbac:config",
```

Comment:
cc @yanavlasov 

I noticed a breakage during out import from OSS because we disable the rbac extension.  The strong dependency on that extension that this change adds breaks compilation in our setup.

I think there's a problem with this test: it depends on an extension but is not using the extension build system macros that remove the test from execution when the relevant extension is removed from the build.  But this test is using the rbac extension as an example in order to test a config loading mechanism, not testing the extension itself, so excluding the test would mean that we lose test coverage for the mechanism you introduced.

Should this test hand craft it's own test-only extension and use that to verify that the extension mechanism works as expected?

Reply:
I could do that, although not sure how valuable that is. I was operating under the assumption that RBAC is a core extension. I do need variable behavior for this test to make sense, so not many existing extensions fit. Feel free to file an issue against me to clean this up.

Reply:
This is technically the list of core extensions. https://github.com/envoyproxy/envoy/blob/dbbcc692621537d0cb4abd54950bb6356b6d01d6/source/extensions/all_extensions.bzl#L19
If you do need to test specific behavior, please create a test filter with the behavior you need. Other wise these tests will impact others who exclude RBAC filter from their build.